### PR TITLE
CMake: Fix building ARM64 on Windows

### PR DESCRIPTION
**Pull request by spycrab**
* Adds two new configurations to ``CMakeSettings.json``: ``Release (arm64)`` and ``Debug (arm64)``
* Skips over ``External/ffmpeg`` for ARM64 builds
* Outputs ARM64 binaries in ``Binary/ARM64`` now
* Fixes a bunch of other build and linking errors

cc: @stenzek

### CODE REVIEW

File: CMakeSettings.json [hunk 1]
```diff
@@ -29,6 +29,52 @@
...
+      "buildCommandArgs": "",
+      "buildRoot": "${workspaceRoot}\\Build\\${name}",
+      "cmakeCommandArgs": "",
+      "variables": [
+        {
+          "name": "Qt5_DIR",
+          "value": "${workspaceRoot}\\Externals\\Qt\\Qt5.14.1\\msvc2019_arm64\\lib\\cmake\\Qt5"
+        },
+        {
+          "name": "CMAKE_SYSTEM_NAME",
```
Comment (stenzek) [inline CMakeSettings.json]: Does this need to be explicitly set, or can it come from the toolchain?

File: CMakeSettings.json [hunk 2]
```diff
@@ -29,6 +29,52 @@
...
+          "value": "Windows"
+        },
+        {
+          "name": "CMAKE_SYSTEM_PROCESSOR",
```
Comment (stenzek) [inline CMakeSettings.json]: Ditto

File: CMakeLists.txt [hunk 3]
```diff
@@ -200,9 +205,11 @@ elseif(_ARCH_64 AND CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|AMD64")
   add_definitions(-D_M_X86_64=1)
   check_and_add_flag(HAVE_SSE2 -msse2)
 elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+  if (NOT MSVC)
```
Comment (stenzek) [inline CMakeLists.txt]: The `_M_ARM` could probably be dropped, I can't see any references to it in the source.

File: CMakeSettings.json (hunk 1 above)
Comment (spycrab) [reply CMakeSettings.json]: Yes it has to be, otherwise the other argument gets ignored as well.

File: CMakeSettings.json (hunk 2 above)
Comment (spycrab) [reply CMakeSettings.json]: See above.

**Pull request merged**